ThroughputTest/README.md

```diff
@@ -0,0 +1,130 @@
+and also allows for running experiments that dynamically adjust some tuning parameters. 
+
+> This tool is meant to be used with Azure Service Bus Premium and not with Azure Service Bus Standard.
+> Azure Service Bus Premium is designed to provide predictable performance, meaning that the results 
+> you measure with this tool are representative of the performance you can expect for your applications. 
+
+The sample assumes that you are generally familiar with the Service Bus .NET SDK and with how to set up 
+namespaces and entities within those namespaces. 
+
+The sample can either send and receive messages from a single instance, or just act as either sender or receiver,
+allowing simulation of different scenarios. 
+
+The sending side supports sending messages singly or in batches, and it supports pipelining of send operations 
+whereby up to a certain number of messages are kept in flight and their acknowledgement is handled asynchronously. 
+You can start one or multiple concurrent senders, and each sender can be throttled by imposing a pause between
+individual send operations.
+
+The receive side supports the "ReceiveAndDelete" and "PeekLock" receive modes, one or multiple concurrent 
+receive loops, single or batch receive operations, and single or batch completion. You can simulate having
+multiple concurrent handlers on a receiver and you can also impose a delay to simulate work.
+
+## What to expect
+
+This sample is specifically designed to test throughput limits of queues and topics/subscriptions in Service Bus 
+namespaces. It does not measure end-to-end latency, meaning how fast messages can pass through Service Bus under 
+optimal conditions. The goals of achieving maximum throughput and the lowest end-to-end latency are fundamentally 
+at odds, as you might know first-hand from driving a car. Either you can go fast on a street with light traffic, or 
+the street can handle a maximum capacity of cars at the same time, but at the cost of individual speed. 
+The goal of this sample is to find out where the capacity limits are. 
+
+As discussed in the [product documentation](https://docs.microsoft.com/azure/service-bus-messaging/message-transfers-locks-settlement), 
+network latency has very significant impact on the achievable throughput. If you are running this sample from your
+development workstation, throughput will be substantially lower than from within an Azure VM. If you want to 
+test limits for a scenario where the Service Bus client will reside inside Azure, you should also run this test 
+inside Azure on a Windows or Linux VM that resiodes within the same region as the Service Bus namespace you 
+want to test.
+
+In an in-region setup, and with ideal parameters, you can achieve send rates exceeding 20000 msg/sec at 1024 bytes per message
+from a single client, meaning that a 1GB queue will fill up in under a minute. Also be aware that receive operations are
```

Comment:
Have you tested that. Most people don't believe that. AMQP unlikely can do that. (At least the tests Jobin did where below 10k. (Some 8k if I remember correctly)